src/_pytest/nodes.py

```diff
@@ -110,45 +111,67 @@ class Node(metaclass=NodeMeta):
         "parent",
         "config",
         "session",
-        "fspath",
+        "fs_path",
         "_nodeid",
         "_store",
         "__dict__",
     )
 
+    name: str
+    config: Config
+    session: "Session"
+    fs_path: Path
+    _nodeid: str
+
     def __init__(
         self,
+        *,
         name: str,
-        parent: "Optional[Node]" = None,
+        parent: Optional["Node"],
         config: Optional[Config] = None,
-        session: "Optional[Session]" = None,
-        fspath: Optional[py.path.local] = None,
```

Comment:
By dropping `fspath` (the `py.path.local` variant) aren't we breaking the API? If so I think a better option would be to still support `fspath`, converting it internally to `fs_path` and issue a deprecation warning.